pilot/platform/cloudfoundry/config.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudfoundry
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"gopkg.in/validator.v2"
+	yaml "gopkg.in/yaml.v2"
+)
+
+type CopilotConfig struct {
+	ServerCACertPath string        `yaml:"server_ca_cert_path" validate:"nonzero"`
+	ClientCertPath   string        `yaml:"client_cert_path" validate:"nonzero"`
+	ClientKeyPath    string        `yaml:"client_key_path" validate:"nonzero"`
+	Address          string        `yaml:"address" validate:"nonzero"`
+	PollInterval     time.Duration `yaml:"poll_interval" validate:"nonzero"`
```

Comment:
what does the `validate:"nonzero"` do here?

Reply:
I believe it validates that the value is provided, so that the property does not fall to a default value.

Reply:
The `validate` struct tag is used by this library: https://github.com/go-validator/validator